File: multimodals-LLMs-Hugging-Face/devzees-ai-server/src/main/java/com/devzees/ai/models/AIModel.java
package com.devzees.ai.models;
public class AIModel {
    private String name;
    private boolean enabled;
    private String icon;

    public AIModel(String name, boolean enabled, String icon) {
        this.name = name;
        this.enabled = enabled;
        this.icon = icon;
    }

    // getters and setters
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public boolean isEnabled() { return enabled; }
    public void setEnabled(boolean enabled) { this.enabled = enabled; }

    public String getIcon() { return icon; }
    public void setIcon(String icon) { this.icon = icon; }
}

File: multimodals-LLMs-Hugging-Face/devzees_ai_client/lib/features/model/ai_model.dart
class AIModel {
  final String name;
  bool enabled;
  final String icon;

  AIModel({required this.name, required this.enabled, required this.icon});

  factory AIModel.fromJson(Map<String, dynamic> json) {
    return AIModel(
      name: json['name'],
      enabled: json['enabled'],
      icon: json['icon'],
    );
  }

  Map<String, dynamic> toJson() => {
    "name": name,
    "enabled": enabled,
    "icon": icon,
  };
}
